File: DataStructures/C/Stack.c
#include <stdio.h>

int MAXSIZE = 8;
int stack[8];
int top = -1;

int isempty()
{

    if (top == -1)
        return 1;
    else
        return 0;
}

int isfull()
{

    if (top == MAXSIZE)
        return 1;
    else
        return 0;
}

int peek()
{
    return stack[top];
}

int pop()
{
    int data;

    if (!isempty())
    {
        data = stack[top];
        top = top - 1;
        return data;
    }
    else
    {
        printf("Could not retrieve data, Stack is empty.\n");
    }
}

int push(int data)
{

    if (!isfull())
    {
        top = top + 1;
        stack[top] = data;
    }
    else
    {
        printf("Could not insert data, Stack is full.\n");
    }
}
int printstack()
{
    while (!isempty())
    {
        int data = pop();
        printf("%d\n", data);
    }
}
int main()
{
    // push items on to the stack
    push(10);
    push(20);
    push(15);
    push(17);
    push(40);
    push(35);

    printf("Element at top of the stack: %d\n", peek());
    printf("Elements in a Stack: \n");
    printstack();

    printf("Stack full: %s\n", isfull() ? "true" : "false");
    printf("Stack empty: %s\n", isempty() ? "true" : "false");

    return 0;
}

File: DataStructures/Java/Stack.java
public class Stack {

	int maxSize;
	long[] array;
	int top;

	public Stack(int i) {
		maxSize = i;
		array = new long[maxSize];
		top = -1;
	}

	public void push(long j) {
		array[++top] = j;
	}

	public long pop() {
		return array[top--];
	}

	public long peek() {
		return array[top];
	}

	public boolean isEmpty() {
		return (top == -1);
	}

	public boolean isFull() {
		return (top == maxSize - 1);
	}

	public static void main(String[] args) {
		Stack stack = new Stack(10);
		stack.push(10);
		stack.push(20);
		stack.push(30);
		stack.push(40);
		stack.push(50);

		while (stack.isEmpty() == false) {
			long value = stack.pop();
			System.out.println(value);
		}
	}

}
